sympy/polys/rings.py

```diff
@@ -2531,6 +2536,71 @@ def compose(f, x, a=None):
 
         return poly
 
+    def coeff_split(self, syms):
+        """
+        Split a polynomial ``self`` into terms based on a set of symbolic
+        indices ``syms``.
```

Comment:
This explanation is not very precise.

I would probably write this as something like:
```
Get the coefficients of a polynomial with respect to the specified ``syms``.

For example, given a polynomial in ``p`` in ``K[x,y,z,t]``, ``p.coeff_split({y, t})`` converts ``p``
to an element of ``K[x, z][y, t]`` and returns the coefficients as elements of ``K[x, z]``.
```